standard/lpdb/count.lua

```diff
@@ -0,0 +1,274 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Count
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lpdb = require('Module:Lpdb')
+
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Team = require('Module:Team')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local Count = {}
+
+---Counts the number of matches played on a wiki - querying lpdb_match2
+---@param args table
+---@return integer?
+function Count.match2(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('match2', lpdbConditions)
+end
+
+---Counts the number of games played on a wiki - querying lpdb_match2
+---@param args table
+---@return integer?
+function Count.match2games(args)
```

Comment:
**maybe** add an option to query via lpdb_match2games directly (doesn't allow some of your base conditions)

Reply:
I'm not sure I follow. I didn't think you could query match2games directly. I get a data type not specified correctly error when trying to query it.

Reply:
ah, its `match2game`

Reply:
it works, see e.g. https://liquipedia.net/starcraft2/index.php?title=Module:AligulacCopyPasteFromLPDB&action=edit#mw-ce-l182 as an example
as mentioned before some of the base conditions are not applicable to match2game queries (since they do not have those keys/fields) and hence would cause errors

Reply:
Have added a function `match2game` which queries match2game directly, and another `match2gamesData` for more specific queries with tier conditions.